MVVM Module/Source/Feed Presentation/FeedViewModel.swift

```
@@ -17,17 +17,26 @@ final class FeedViewModel {
 	var title: String {
         Localized.Feed.title
 	}
-
+    
+    var errorMessage: String {
+        Localized.Feed.errorMessage
+    }
+    
 	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
+    var onFailure: Observer<Error>?
```

`onFailure: Observer<Error> ` could be `onErrorStateChange: Observer<String?>`, so the ViewModel decides when to show/hide errors (instead of the view controller). It's much easier to manage view state when all changes are centralized in the ViewModel.

For example, in the `loadFeed()` method, the ViewModel can call  `onErrorStateChange(.none)` to hide the error before loading the feed.

Moreover, notice the type is optional string: `Observer<String?>` because the ViewModel should decide which formatted string to show in the UI. For example, if can call `onErrorStateChange(Localized.Feed.errorMessage)`